src/trait_/partial_eq.rs

```diff
@@ -65,6 +104,11 @@ impl TraitImpl for PartialEq {
 				}
 				// If there is only one variant and it's empty or if the struct is empty, simple
 				// return `true`.
+				Item::Enum { variants, .. }
+					if variants.len() == 1 && variants[0].is_incomparable() =>
```

Comment:
This should probably also cover the case if there is more then one variant and all of them are incomparable.

Reply:
yes, good catch